src/change_res.cpp

```diff
@@ -56,3 +68,193 @@ void ChangeRes(int, int, int, int)
 //{
 
 //}
+
+void UpdateInternalRes()
+{
+    int req_w = g_config.InternalW;
+    int req_h = g_config.InternalH;
+    if((!g_compatibility.free_level_res && !LevelSelect && !GameMenu)
+        || (!g_compatibility.free_world_res && LevelSelect && !GameMenu))
+    {
+        if((req_w != 0 && req_w < 800) || (req_h != 0 && req_h < 600))
+        {
+            req_w = 800;
+            req_h = 600;
+
+#ifdef THEXTECH_FIXED_RES
+            PlaySoundMenu(SFX_BowserKilled);
+            MessageText = "Sorry! The requested compatibility mode was not enabled because your copy of TheXTech was not built to support a resolution of 800x600.";
+            PauseGame(PauseCode::Message);
+            MessageText.clear();
+#endif
+        }
+    }
+
+#ifndef THEXTECH_FIXED_RES
+    if(req_w == 0 || req_h == 0)
+    {
+        int int_w, int_h;
+
+        XWindow::getWindowSize(&int_w, &int_h);
+
+        int orig_int_h = int_h;
+
+        // set internal height first
+        if(req_h != 0)
+        {
+            int_h = req_h;
+        }
+        else if(g_videoSettings.scaleMode == SCALE_FIXED_05X)
+        {
+            int_h *= 2;
+        }
+        else if(g_videoSettings.scaleMode == SCALE_FIXED_2X)
+        {
+            int_h /= 2;
+        }
+        else if(g_videoSettings.scaleMode == SCALE_DYNAMIC_INTEGER)
+        {
+            if(int_h >= 600)
+            {
+                // constrains height to be in the 600-720p range
+                int scale_factor = int_h / 600;
+                int_h /= scale_factor;
+            }
+        }
+
+        // minimum height constraint
+        if(int_h < 320)
+            int_h = 320;
+
+        // maximum height constraint
+        if(int_h > 720 && req_h <= 720)
+            int_h = 720;
+
+        // now, set width based on height and scaling mode
+        if(g_videoSettings.scaleMode == SCALE_FIXED_05X)
+        {
+            int_w = int_w * 2;
+        }
+        else if(g_videoSettings.scaleMode == SCALE_FIXED_1X)
+        {
+            int_w = int_w;
+        }
+        else if(g_videoSettings.scaleMode == SCALE_FIXED_2X)
+        {
+            int_w = int_w / 2;
+        }
+        else if(g_videoSettings.scaleMode == SCALE_DYNAMIC_INTEGER)
```

Comment:
Also redundant braces for single-line code blocks.

Reply:
Same case as above, the last clause is multiline and I want them to be balanced.